libdnf5/base/goal.cpp

```diff
@@ -509,6 +522,8 @@ GoalProblem Goal::Impl::add_specs_to_goal(base::Transaction & transaction) {
                         base, query, settings.advisory_filter.value(), cfg_main.get_obsoletes_option().get_value());
                 }
 
+                remove_older_versions(query);
```

Comment:
May I ask you to reference the bug or write a note why it is important?

Reply:
Sure. Added to `remove_older_versions()` function description.